### DISCUSSION TITLE
Split `File` authentication backend into two files

### DISCUSSION THREAD

**Discussion by m-theisen** [authelia/authelia#4638] [Ideas]
# Feature Request

## Description
It would be great if the `file`-backend for users  would support splitting the "database" into two files:
1) "Immutable"/admin information: user name, e-mail,  groups
2) "Mutable"/user information: display name, password

## Use Case
This would allow for an easier "git-ops" driven approach to managing authelia. An example for a Kubernetes deployment via Flux/Argo CD:
- Specify the "immutable" part as a (read-only) `configMap` inside the git source which is automatically applied via Flux/Argo CD
- Keeping the "mutable" part in a `volumeClaim` to allow the users to freely change their password and display name without any admin interaction

## Side Notes
A further improvement in this context could be (I have not thought of any security implications) to set an automatic secure random password for a new user (added to the "immutable" file)  and send the "recover password  mail" to the new user, so that they can set their desired password.\
Like this a new user would be notified as soon as their account is ready and an admin does not need to set a default password. The latter not being "easy" in the scenario where the password file sits in a `volumeClaim`. 

Another thing to consider would be allowing the "mutable" part to resign in the SQL backend (once it is implemented), this will probably make the implementation more complicated though.

P.S.: Thanks for this great project

**Comment by james-d-elliott** [comment]
We'll add this via the authelia UI, unfortunately this idea is what I'd consider over-engineering and I'm actually firmly against it. 

Usernames and emails will likely be something users can adjust in the future, and we're likely to support registration functionality. Both of which require full write access to the entire file. This is a minimum requirement of this backend.
